### PR TITLE
New Terraform sample for a Quickstart for Azure NAT Gateway that creates a NAT gateway and supporting resources

### PR DESCRIPTION
**Pull request by asudbring**
…rresponding network, subnet and public IP addresses

### CODE REVIEW

File: quickstart/101-nat-gateway-create/main.tf [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+resource "azurerm_network_interface_security_group_association" "example" {
+  network_interface_id      = azurerm_network_interface.my_terraform_nic.id
+  network_security_group_id = azurerm_network_security_group.my_terraform_nsg.id
+}
+
+# Generate random text for a unique storage account name
+resource "random_id" "random_id" {
+  keepers = {
+    # Generate a new ID only when a new resource group is defined
+    resource_group = azurerm_resource_group.rg.name
```
Comment (lonegunmanb) [inline quickstart/101-nat-gateway-create/main.tf:103]: Why we need such a `keepers` here?

**Review by lonegunmanb — COMMENTED** [review]
Hi @asudbring almost LGTM but only one question.

File: quickstart/101-nat-gateway-create/main.tf (hunk 1 above)
Comment (asudbring) [reply quickstart/101-nat-gateway-create/main.tf:103]: That's a good question. I copied the VM install from another sample unchanged.  Is this a resource I can remove safely?  Thanks!

Comment (asudbring) [reply quickstart/101-nat-gateway-create/main.tf:103]: @lonegunmanb 
That highlighted part it looks like is required for the storage account that is created to store diagnostics for the virtual machine and looks like it is needed.

Thanks!

Comment (TomArcherMsft) [reply quickstart/101-nat-gateway-create/main.tf:103]: @lonegunmanb @asudbring In this context, I believe that "keepers" ensures that if the resource group name changes, the storage account will be created with a new name.

**Pull request merged**